src/test_workflow/integ_test/topology.py

```diff
@@ -0,0 +1,109 @@
+# Copyright OpenSearch Contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+import os
+from contextlib import contextmanager
+from pathlib import Path
+from typing import Any, Generator, List
+
+from manifests.bundle_manifest import BundleManifest
+from test_workflow.dependency_installer import DependencyInstaller
+from test_workflow.integ_test.local_test_cluster import LocalTestCluster
+from test_workflow.test_recorder.test_recorder import TestRecorder
+
+
+class Topology:
+
+    def __init__(
+        self,
+        topology_config: list,
+        dependency_installer: DependencyInstaller,
+        work_dir: Path,
+        component_name: str,
+        additional_cluster_config: dict,
+        bundle_manifest: BundleManifest,
+        security_enabled: bool,
+        component_test_config: str,
+        test_recorder: TestRecorder
```

Comment:
Half of these fields do not belong in `Topology`. Things that are properties of a topology (e.g. topology configuration) should be parameters to `init`, and things that are needed to `create_clusters` should be parameters to `create` and then `create_clusters`.